ProcessOptimizer/plots.py

```diff
@@ -1,2382 +1,2493 @@
+                            c="k",
+                            s=20,
+                            marker="D",
+                            zorder=0,
+                        )
+                    else:
+                        # Simply show the mean value
+                        ax[i, i].scatter(
+                            xi,
+                            zi,
+                            c="red",
+                            s=80,
+                            marker="_",
+                            zorder=1,
+                        )
+                        # Also add highlight the best/expected minimum
+                        ax[i, i].scatter(
+                            minimum[i],
+                            zi[int(minimum[i])],
+                            c="k",
+                            s=20,
+                            marker="D",
+                            zorder=0,
+                        )
+
+                # For non-categoric factors
+                else:
+                    ax[i, i].set_xlim(np.min(xi), np.max(xi))
+                    ax[i, i].axvline(
+                        minimum[i], linestyle="--", color="k", lw=1)
+                    if show_confidence:
+                        ax[i, i].fill_between(xi,
+                                              y1=(zi - 1.96*stddevs),
+                                              y2=(zi + 1.96*stddevs),
+                                              alpha=0.5,
+                                              color="green",
+                                              edgecolor="green",
+                                              linewidth=0.0,
+                                              )
+                    else:
+                        ax[i, i].plot(
+                            xi,
+                            zi,
+                            color="red",
+                            lw=1,
+                            zorder=0,
+                        )
+
+                ymin = limits["val_1d_min"]-abs(limits["val_1d_min"])*.02
+                ymax = limits["val_1d_max"]+abs(limits["val_1d_max"])*.02
+                ax[i, i].set_ylim(ymin, ymax)
+
+            # lower triangle
+            elif i > j:
+                _2d_dependency_plot(
+                    data=plots_data[i][j],
+                    axes=ax[i][j],
+                    samples=(samples[:, j], samples[:, i]),
+                    highlighted=(minimum[j], minimum[i]),
+                    limits=limits,
+                    options=plot_options
+                )
+                if [i, j] == [1, 0]:
+                    # Add a colorbar for the 2D plot value scale
+                    import matplotlib as mpl
+
+                    norm_color = mpl.colors.Normalize(
+                        vmin=limits["z_min"], vmax=limits["z_max"]
+                    )
+                    cb = ax[0][-1].figure.colorbar(
+                        mpl.cm.ScalarMappable(
```

Comment:
If you want to break the single line formatting here, we should do so for each argument to ScalarMappable and ensure the closing parenthesis ends on its own line.